thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/email/filter/BaseAlertFilter.java

```diff
@@ -23,23 +28,34 @@ public void setParameters(Map<String, String> parameterSetting) {
     Class c = this.getClass();
     for (String fieldName : getPropertyNames()) {
       Double value = null;
+      String strVal = null;
       // Get user's value for the specified field
       if (parameterSetting.containsKey(fieldName)) {
-        value = Double.parseDouble(parameterSetting.get(fieldName));
+        String fieldVal = parameterSetting.get(fieldName);
+        if (NumberUtils.isNumber(fieldVal)) {
+          value = Double.parseDouble(parameterSetting.get(fieldName));
+        } else {
+          strVal = fieldVal;
+        }
       } else {
         // If user's value does not exist, try to get the default value from Class definition
         try {
-          Field field = c.getDeclaredField("DEFAULT_" + fieldName.toUpperCase());
+          Field field = c.getDeclaredField(fieldName);
           boolean accessible = field.isAccessible();
           field.setAccessible(true);
-          value = Double.parseDouble((String) field.get(this));
+          Object object = field.get(this);
```

Comment:
Why the block at lines 35-39 and the blocks at lines 46-51 are different? It would be easier to maintain if we apply the same logic to populate the variables.